### DISCUSSION TITLE
Error code E401

### DISCUSSION THREAD

**Discussion by DeepakMalkan** [iTwin/itwinjs-core#1424] [Q&A] [answered]
I'm trying out the steps on: https://developer.bentley.com/tutorials/web-application-quick-start/

I get an authentication error in step 4, when I run the 'npx' command. What am I missing?

My .env file is in the same folder from where I run the 'npx' command.

The 'npx' command generated the following log file:

0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   '--save',
1 verbose cli   '--save-exact',
1 verbose cli   '--loglevel',
1 verbose cli   'error',
1 verbose cli   'react',
1 verbose cli   'react-dom',
1 verbose cli   '@bentley/react-scripts',
1 verbose cli   '@bentley/cra-template-itwin-viewer'
1 verbose cli ]
2 info using npm@6.14.13
3 info using node@v14.17.0
4 verbose npm-session 4c039d2b712c742c
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 http fetch GET 401 https://pkgs.dev.azure.com/bentleycs/_packaging/Packages/npm/registry/@bentley%2freact-scripts 132ms
8 silly fetchPackageMetaData error for @bentley/react-scripts@latest Unable to authenticate, need: Basic realm="https://pkgsprodcus1.pkgs.visualstudio.com/"
9 http fetch GET 304 https://registry.npmjs.org/react 154ms (from cache)
10 silly pacote tag manifest for react@latest fetched in 169ms
11 http fetch GET 304 https://registry.npmjs.org/react-dom 162ms (from cache)
12 silly pacote tag manifest for react-dom@latest fetched in 165ms
13 http fetch GET 401 https://pkgs.dev.azure.com/bentleycs/_packaging/Packages/npm/registry/@bentley%2fcra-template-itwin-viewer 263ms
14 silly fetchPackageMetaData error for @bentley/cra-template-itwin-viewer@latest Unable to authenticate, need: Basic realm="https://pkgsprodcus1.pkgs.visualstudio.com/"
15 timing stage:rollbackFailedOptional Completed in 2ms
16 timing stage:runTopLevelLifecycles Completed in 308ms
17 verbose stack Error: Unable to authenticate, need: Basic realm="https://pkgsprodcus1.pkgs.visualstudio.com/"
17 verbose stack     at C:\Users\Deepak.Malkan\AppData\Roaming\nvm\v12.20.1\node_modules\npm\node_modules\npm-registry-fetch\check-response.js:107:17
17 verbose stack     at processTicksAndRejections (internal/process/task_queues.js:95:5)
18 verbose statusCode 401
19 verbose pkgid @bentley/react-scripts@latest
20 verbose cwd D:\deepak\source\itwinjs\developer-starter\start-app
21 verbose Windows_NT 10.0.18363
22 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "--save" "--save-exact" "--loglevel" "error" "react" "react-dom" "@bentley/react-scripts" "@bentley/cra-template-itwin-viewer"
23 verbose node v14.17.0
24 verbose npm  v6.14.13
25 error code E401
26 error Unable to authenticate, need: Basic realm="https://pkgsprodcus1.pkgs.visualstudio.com/"
27 verbose exit [ 1, true ]

**Comment by calebmshafer** [comment, marked as answer]
@DeepakMalkan it looks like you're hitting an issue with the private Bentley registry. You will want to follow the steps on the internal wiki to update your auth token in the `.npmrc` file.  I'll follow up by email with the link.
